Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.h

```diff
@@ -66,11 +67,25 @@ class SOFA_COMPONENT_CONSTRAINT_LAGRANGIAN_SOLVER_API GenericConstraintProblem :
     void freeConstraintResolutions();
     void solveTimed(SReal tol, int maxIt, SReal timeout) override;
 
+    /// Projective Gauss Seidel method building the compliance matrix
     void gaussSeidel(SReal timeout=0, GenericConstraintSolver* solver = nullptr);
+    /// Projective Gauss Seidel unbuilt method
     void unbuiltGaussSeidel(SReal timeout=0, GenericConstraintSolver* solver = nullptr);
+    /// Method from:
+    /// A nonsmooth nonlinear conjugate gradient method for interactive contact force problems
+    /// - 2010, Silcowitz, Morten and Niebe, Sarah and Erleben, Kenny
+    void NNCG(GenericConstraintSolver* solver = nullptr, int iterationNewton = 1);
 
     int getNumConstraints();
     int getNumConstraintGroups();
+
+protected:
+protected:
+    sofa::linearalgebra::FullVector<double> lam;
```

Comment:
I would prefer a more explicit name.
Also, add the `m_` prefix to protected or private class members